app/api/__init__.py

```diff
@@ -15,7 +16,14 @@
 
 # users
 api.route(UserList, 'user_list', '/users')
-api.route(UserDetail, 'user_detail', '/users/<int:id>')
+api.route(UserDetail, 'user_detail', '/users/<int:id>', '/notifications/<int:notification_id>/user')
```

Comment:
Please see the implementation of EventDetail class. There is a method ```before_get_object```. You have to implement this on UserDetail as well. So that it can get related id of user of that specific notification.